docs(orderBy): improve reverse order example

rjenks opened the pull request:
Changed the second example to demonstrate typical sorting behavior (like Finder and Win Explorer) where the first click on a column sorts alpha and the second click on an already sorted column sorts in reverse alpha.  Also shows a simple sort indicator automated using ngShow.

rjenks committed:
docs(orderBy): describe your change...

Changed the second example to demonstrate typical sorting behavior (like Finder and Win Explorer) where the first click on a column sorts alpha and the second click on an already sorted column sorts in reverse alpha.  Also shows a simple sort indicator automated using ngShow.

rjenks commented:
I signed the CLA

gkalpak commented on src/ng/filter/orderBy.js:
```diff
@@ -92,10 +92,18 @@
-             <th><a href="" ng-click="predicate = 'phone'; reverse=!reverse">Phone Number</a></th>
-             <th><a href="" ng-click="predicate = 'age'; reverse=!reverse">Age</a></th>
+             <th>
+               <a href="" ng-click="predicate == 'name' ? reverse = !reverse : reverse = false; predicate='name';">Name</a>
```
IMO, it would be clearer like this (here and below):

```js
reverse = (predicate === 'name') ? !reverse : false; predicate = 'name'
```

gkalpak commented on src/ng/filter/orderBy.js:
```diff
@@ -92,10 +92,18 @@
-             <th><a href="" ng-click="predicate = 'age'; reverse=!reverse">Age</a></th>
+             <th>
+               <a href="" ng-click="predicate == 'name' ? reverse = !reverse : reverse = false; predicate='name';">Name</a>
+               <span ng-show="predicate == 'name'"><span ng-show="reverse">&#9660;</span><span ng-show="!reverse">&#9650;</span></span>
```
Here again, I would prefer:

```html
<span ng-show="predicate === 'name'">{{ reverse ? '#&9660' : '#&9650' }}</span>
```

gkalpak commented:
I left a couple of minor (style) comments, but in general I like this.

petebacondarwin commented:
Personally I would rather have this logic in the controller.

gkalpak commented:
I didn't rhink there would be a controller for such simple example, but since there is one already, I too agree it is a better practice to have it on the controller.

rjenks commented:
How about this:

```html
<script>
  angular.module('orderByExample', [])
    .controller('ExampleController', ['$scope', function($scope) {
      $scope.friends =
          [{name:'John', phone:'555-1212', age:10},
           {name:'Mary', phone:'555-9876', age:19},
           {name:'Mike', phone:'555-4321', age:21},
           {name:'Adam', phone:'555-5678', age:35},
           {name:'Julie', phone:'555-8765', age:29}];
      $scope.predicate = 'age';
      $scope.reverse = true;
      $scope.order = function(predicate) {
          $scope.reverse = ($scope.predicate === predicate) ? !$scope.reverse : false;
          $scope.predicate = predicate;
      };
    }]);
</script>
<style type="text/css">
    SPAN.sortorder:after {
        content: '\25b2';
    }

    SPAN.sortorder.reverse:after {
        content: '\25bc';
    }
</style>
<div ng-controller="ExampleController">
  <pre>Sorting predicate = {{predicate}}; reverse = {{reverse}}</pre>
  <hr/>
  [ <a href="" ng-click="predicate=''">unsorted</a> ]
  <table class="friend">
    <tr>
      <th>
        <a href="" ng-click="order('name')">Name</a>
        <span class="sortorder" ng-show="predicate === 'name'" ng-class="{reverse:reverse}"></span>
      </th>
      <th>
        <a href="" ng-click="order('phone')">Phone Number</a>
        <span class="sortorder" ng-show="predicate === 'phone'" ng-class="{reverse:reverse}"></span>
      </th>
      <th>
        <a href="" ng-click="order('age')">Age</a>
        <span class="sortorder" ng-show="predicate === 'age'" ng-class="{reverse:reverse}"></span>
      </th>
    </tr>
    <tr ng-repeat="friend in friends | orderBy:predicate:reverse">
      <td>{{friend.name}}</td>
      <td>{{friend.phone}}</td>
      <td>{{friend.age}}</td>
    </tr>
  </table>
</div>
```

gkalpak commented:
I like it much better. I wonder if it would make sense to go one step further and `ngRepeat` over the `<th>` (instead of having 3 hard-coded elements with almost identical content).
Not sure if it make the example too complex to focus on the `orderByFilter` functionality, though.

rjenks commented:
Either way, but I also wonder if that may complicate the example.  Also, if we did this two this example then it should probably also be done to the other examples that are on this same doc page.  I am happy to make the changes and update the pull request as you see fit.

gkalpak commented:
I'll leave it to @petebacondarwin's judgement :)

petebacondarwin commented:
I think that @rjenks latest idea is good enough. Do you want to update the PR @rjenks? Then do you want to merge it @gkalpak? Thanks guys

rjenks committed:
Changes to sort order example per discussion

Moved order logic into controller.
Added CSS for rendering sort indicator to simplify example

gkalpak commented:
@rjenks, could you squash the commits into one (with an appropriate commit message) ?

petebacondarwin commented:
@gkalpak - at this stage it is probably simplest to squash and tweak the commit message ourselves when merging. Generally, even if the @rjenks squashes, etc, we will still have to rebase again onto master.

gkalpak commented:
@petebacondarwin, OK, will squash upon merging then :)

petebacondarwin commented:
Thanks!
